src/plugins/discover/public/application/doc/single_doc_route.tsx

```diff
@@ -31,11 +32,17 @@ export interface DocUrlParams {
 
 const SingleDoc = ({ id }: SingleDocRouteProps) => {
   const services = useDiscoverServices();
-  const { chrome, timefilter } = services;
+  const { chrome, timefilter, core } = services;
 
   const { indexPatternId, index } = useParams<DocUrlParams>();
   const breadcrumb = useMainRouteBreadcrumb();
 
+  useExecutionContext(core.executionContext, {
+    type: 'application',
+    page: 'single-doc',
+    id: indexPatternId || '',
```

Comment:
Can we add something more meaningful than an empty string? Like `unknown`?

Reply:
This is actually a copy paste error.
`indexPatternId` is never empty.